### DISCUSSION TITLE
How can I parse a series of flags/tokens into a struct?

### DISCUSSION THREAD

**Discussion by ip1981** [winnow-rs/winnow#594] [Q&A] [answered]
Parse `"foo,bar,..."` (maybe more foos and bars) into a structure:

```rust
struct A {
  foo: bool,
  bar: bool
}
```

**Comment by epage** [comment]
So you have a comma-delimited list of field names and if a field is present, you want to mark the field as `true`?

Some ideas
- Imperatively loop while you can parse `opt(',')` and then capture the next `alpha1` and `match` it for the field, initializing it
  - I had hoped that `separated_foldl1` would handle that but it assumes the parser output and the `fold`s output are the same type.  That should likely be improved
- Tokenize into a `enum AField` and then have a function that turns that into an `A`.

**Comment by ip1981** [comment, marked as answer]
I've done this with `separated_foldl1` and implementing `AddAssign` (I needed this trait in other place, otherwise I'd use `Add`) for the structure. Something like this (`parse_foo`, `parse_bar` return the same structure but with different fields defined):
```rust
            separated_foldl1(
                cut_err(alt((parse_foo, parse_bar))),
                (multispace0, ',', multispace0),
                |mut l, _, r| {
                    l += r;
                    l
                },
            )
```
